inbc-program/inbc/parser/source_app_parser.py

```diff
@@ -47,13 +56,16 @@ def application_remove(args: argparse.Namespace) -> str:
     manifest = ('<?xml version="1.0" encoding="utf-8"?>' +
                 '<manifest><type>source</type>' +
                 '<applicationSource>' +
-                '<remove><gpg>'
-                f'{create_xml_tag(arguments, "keyname")}' +
-                '</gpg><repo>' +
-                f'{create_xml_tag(arguments, "filename")}'
-                '</repo>'
-                '</remove></applicationSource>' +
-                '</manifest>')
+                '<remove>')
+
+    if args.gpgKeyName:
+        manifest += f'<gpg>{create_xml_tag(arguments, "keyname")}</gpg>'
```

Comment:
**Suggestion:** Avoid using f-strings inside expressions where curly braces are used for XML tags, as it might cause confusion. Use format method or concatenate the strings for clarity. [readability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        manifest += '<gpg>' + create_xml_tag(arguments, "keyname") + '</gpg>'
```